front/pages/api/w/[wId]/vaults/[vId]/data_sources/index.ts

```diff
@@ -0,0 +1,542 @@
+  configuration: ConnectorConfigurationTypeSchema,
+});
+
+const PostStaticDataSourceRequestBodySchema = t.type({
+  name: t.string,
+  description: t.string,
+  assistantDefaultSelected: t.boolean,
+});
+
+type PostVaultDataSourceResponseBody = {
+  dataSourceView: DataSourceViewType;
+};
+
+async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<WithAPIErrorResponse<PostVaultDataSourceResponseBody>>,
+  auth: Authenticator
+): Promise<void> {
+  const owner = auth.workspace();
+  const plan = auth.plan();
+  const user = auth.user();
+
+  if (!owner || !plan || !user) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "workspace_not_found",
+        message: "The workspace you requested was not found.",
+      },
+    });
+  }
+
+  const vault = await VaultResource.fetchById(auth, req.query.vId as string);
+
+  if (
+    !vault ||
+    (!auth.isAdmin() && !auth.hasPermission([vault.acl()], "write"))
+  ) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "vault_not_found",
+        message: "The vault you requested was not found.",
+      },
+    });
+  }
+
+  const dataSourceEmbedder = owner.defaultEmbeddingProvider ?? "openai";
+  const embedderConfig = EMBEDDING_CONFIGS[dataSourceEmbedder];
+  const coreAPI = new CoreAPI(config.getCoreAPIConfig(), logger);
+
+  switch (req.method) {
+    case "POST": {
+      const isStaticDataSource = typeof req.body.provider !== "string";
+
+      /**
+       * Posting a static data source
+       * (folder)
+       */
+      if (isStaticDataSource) {
+        const bodyValidation = PostStaticDataSourceRequestBodySchema.decode(
+          req.body
+        );
+        if (isLeft(bodyValidation)) {
+          const pathError = reporter.formatValidationErrors(
+            bodyValidation.left
+          );
+          return apiError(req, res, {
+            status_code: 400,
+            api_error: {
+              type: "invalid_request_error",
+              message: `Invalid request body to post a static data source: ${pathError}`,
+            },
+          });
+        }
+
+        const { name, description, assistantDefaultSelected } =
+          bodyValidation.right;
+
+        if (name.startsWith("managed-")) {
+          return apiError(req, res, {
+            status_code: 400,
+            api_error: {
+              type: "invalid_request_error",
+              message: "The data source name cannot start with `managed-`.",
+            },
+          });
+        }
+        if (!isDataSourceNameValid(name)) {
+          return apiError(req, res, {
+            status_code: 400,
+            api_error: {
+              type: "invalid_request_error",
+              message:
+                "Data source names must only contain letters, numbers, and the characters `._-`, and cannot be empty.",
+            },
+          });
+        }
+        const dataSources = await getDataSources(auth);
```

Comment:
Let's avoid using this legacy method  (it will be killed once we release vault) and instead we can do `DataSourceResource.listByWorkspace(auth)`.